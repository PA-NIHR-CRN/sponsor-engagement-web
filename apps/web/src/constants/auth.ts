/**
 * Identifer of our custom next-auth provider
 */
export const AUTH_PROVIDER_ID = 'oidc'

/**
 * Name of our custom next-auth provider
 */
export const AUTH_PROVIDER_NAME = 'OIDC'

/**
 * Auth type of our custom next-auth provider
 */
export const AUTH_PROVIDER_TYPE = 'oauth'

/**
 * Fallback session expiry to use when the NEXTAUTH_SESSION_EXPIRY env var is not provided
 * This is specifically for next-auth's JWT session expiration and not IDG's access token expiration
 */
export const AUTH_SESSION_EXPIRY_FALLBACK = 2592000 // 30 days

/**
 * Local roles associated with a user
 */
export enum Roles {
  SponsorContact = 1,
  ContactManager = 2,
}

export const PASSWORD_MIN_LENGTH = 12

export const ODP_ROLE = 'ODP_SponsorEngagementTool'
