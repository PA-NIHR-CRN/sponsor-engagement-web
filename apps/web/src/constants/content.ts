/**
 * Number of studies to show per page
 */
export const PER_PAGE = 10

/**
 * Ser title prefix to appear on the browser tab
 */
export const SERVICE_NAME = 'Assess progress of studies'

/**
 * Date format to use across FRF
 */
export const DATE_FORMAT = 'D MMMM YYYY'
