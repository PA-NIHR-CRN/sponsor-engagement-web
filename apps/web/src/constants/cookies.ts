/**
 * Name of cookie which stores whether user accepted/rejected additional cookies
 */
export const SE_GDPR_COOKIE_NAME = 'SEConsentGDPR'

/**
 * Value of GDPR consent cookie when user has accepted additional cookies
 */
export const SE_GDPR_COOKIE_ACCEPT_VALUE = 'Accept'

/**
 * Value of GDPR consent cookie when user has rejected additional cookies
 */
export const SE_GDPR_COOKIE_REJECT_VALUE = 'Reject'

/**
 * Number of months until GDPR consent cookie expires
 */
export const SE_GDPR_COOKIE_EXPIRY_MONTHS = 12

/**
 * Name of custom event for when a user changes their cookie settings
 */
export const COOKIE_SETTINGS_CHANGE_EVENT = 'cookieSettingsChanged'
