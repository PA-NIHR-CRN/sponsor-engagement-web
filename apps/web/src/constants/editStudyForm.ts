export const PAGE_TITLE = 'Update study data'
export const GENERIC_STUDIES_GUIDANCE_TEXT =
  'Changes to the study status, the key dates and recruitment targets will be communicated to RDN, where possible, your changes will update the study record automatically in CPMS, other changes might be subject to review by the RDN team.'

enum StudyStatus {
  InSetup = 'In setup',
  OpenToRecruitment = 'Open to recruitment',
  ClosedFollowUp = 'Closed to Recruitment, In Follow Up',
  Closed = 'Closed',
  Withdrawn = 'Withdrawn',
  Suspended = 'Suspended',
}

export const studyStatuses = [
  {
    name: 'In setup',
    description: 'Not yet open to recruitment.',
    id: '1',
    value: StudyStatus.InSetup,
  },
  {
    name: 'Open to recruitment',
    description: 'Ready (open) to recruit participants in at least one UK site. Provide an actual opening date below.',
    id: '2',
    value: StudyStatus.OpenToRecruitment,
  },
  {
    name: 'Closed, in follow-up',
    description:
      'Ongoing, (i.e. participants are being treated or observed), but recruitment is complete. Provide an actual closure date below.',
    id: '3',
    value: StudyStatus.ClosedFollowUp,
  },
  {
    name: 'Closed',
    description:
      'Completed recruitment and any subsequent patient related activities (follow up). Provide an actual closure date below.',
    id: '4',
    value: StudyStatus.Closed,
  },
  {
    name: 'Withdrawn',
    description: 'Withdrawn during the setup phase and will not be opening to recruitment in the UK.',
    id: '5',
    value: StudyStatus.Withdrawn,
  },
  {
    name: 'Suspended',
    description: 'Recruitment of participants has halted, but may resume. Provide an estimated re-opening date below.',
    id: '6',
    value: StudyStatus.Suspended,
  },
]
