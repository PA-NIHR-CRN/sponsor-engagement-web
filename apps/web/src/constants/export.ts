/**
 * File name for the downloaded export
 */
export const FILE_NAME = 'studies.xlsx'

/**
 * Helper/intro text for the excel document
 */
export const HELPER_TEXT = `This download is a snapshot of all the information held within the Sponsor Engagement Tool for the sponsor/delegate organisation. Once exported, the information within this document may become out of date and is no longer guaranteed to be an accurate copy of data held by the live Sponsor Engagement Tool. 

This export was created on %s.

Please note that the security of the information contained in this document is the responsibility of the sponsor who requested the export as per the tool's Terms and Conditions (https://sites.google.com/nihr.ac.uk/crncc-policies/set/terms-and-conditions) and Privacy Notice (https://sites.google.com/nihr.ac.uk/crncc-policies/set/privacy-notice).

Data in this document cannot be imported back into the Sponsor Engagement Tool. 

The last columns in the spreadsheet are provided only to assist with local data gathering for new study assessments. For example; if there is an update to a study status, planned or actual milestone dates or targets, information can be added to the ‘additional information’ column. Once collated, information then has to be entered into the live Sponsor Engagement Tool (https://assessmystudy.nihr.ac.uk) by the sponsor / delegate representative. 

If any further information, advice or support is required, please contact supportmystudy@nihr.ac.uk`

/** Pink colour for editable column headings */
export const PINK_FILL = { type: 'pattern', pattern: 'solid', fgColor: { argb: 'F6D7D2' } } as const

/** Red text colour */
export const RED_TEXT = { font: { color: { argb: 'B40000' } } } as const

export const Column = {
  SponsorAssessment: 'U',
  AdditionalInformation: 'V',
}
