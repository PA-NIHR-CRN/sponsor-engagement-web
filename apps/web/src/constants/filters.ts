import type { FilterKey } from '../@types/filters'

// Filters which can be modified from the side panel/selected filters panel
export const SELECTABLE_FILTERS: FilterKey[] = ['q']
