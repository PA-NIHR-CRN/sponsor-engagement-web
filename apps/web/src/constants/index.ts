export * from './filters'
export * from './content'
export * from './identifiers'
export * from './auth'
