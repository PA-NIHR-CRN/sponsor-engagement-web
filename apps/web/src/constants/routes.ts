/**
 * Signin page that unauthenticated users are redirected to
 */
export const SIGN_IN_PAGE = '/auth/signin'

/**
 * List of studies that users with sponsor contact and contact manager roles can access
 */
export const STUDIES_PAGE = '/studies'

/**
 * List of organisations that users with contact manager role can access
 */
export const ORGANISATIONS_PAGE = '/organisations'

/**
 * Error page for internal server errors
 */
export const ERROR_PAGE_500 = '/500'

/**
 * Error page for page not found
 */
export const ERROR_PAGE_404 = '/404'
