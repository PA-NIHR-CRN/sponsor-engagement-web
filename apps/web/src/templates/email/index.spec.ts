import { EXTERNAL_CRN_TERMS_CONDITIONS_URL, EXTERNAL_CRN_URL, SUPPORT_PAGE } from '../../constants/routes'
import emailTemplates from '.'

const templates = Object.keys(emailTemplates).filter((name) => name !== 'index.js')

test.each(templates)('Email template %p displays correctly', (key) => {
  const template = emailTemplates[key] as (data: Record<string, string>) => string
  expect(
    template({
      crnLink: EXTERNAL_CRN_URL,
      organisationName: 'Mocked org name',
      requestSupportLink: `http://localhost:3000${SUPPORT_PAGE}`,
      signInLink: 'http://localhost:3000/auth/signin',
      termsAndConditionsLink: EXTERNAL_CRN_TERMS_CONDITIONS_URL,
    })
  ).toMatchSnapshot()
})
