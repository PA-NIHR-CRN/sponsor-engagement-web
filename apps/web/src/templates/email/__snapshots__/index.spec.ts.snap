// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email template "contact-assigned.html.hbs" displays correctly 1`] = `
"<html lang='en'>
  <body>
    <div>
      <h1>NIHR CRN has invited you to assess the progress of your studies</h1>

      <p>Hello,</p>

      <p>
        You have been nominated as a contact for
        Mocked org name
        studies on the NIHR Clinical Research Network’s (CRN) research portfolio.
      </p>

      <p>
        The
        <a href='https://www.nihr.ac.uk/explore-nihr/support/clinical-research-network.htm'>NIHR CRN</a>
        supports Sponsors and their delegates to deliver high-quality research, for the benefit of patients, the public
        and health and care organisations across England.
      </p>

      <p>
        The NIHR CRN tracks the progress of research studies in its portfolio using data provided by Sponsors or their
        delegates on behalf of the Department of Health and Social Care.
      </p>

      <p>
        NIHR CRN asks Sponsors or their delegates to review these studies and provide regular assessments of how the
        study is progressing.
      </p>

      <p>
        Please sign in to the Sponsor Engagement Tool to provide updates on studies.
      </p>

      <a
        href='http://localhost:3000/auth/signin'
        style='
          box-sizing: border-box;
          display: inline-block;
          position: relative;
          margin-top: 0;
          margin-right: 0;
          margin-left: 0;
          margin-bottom: 22px;
          padding: 8px 10px 7px;
          border: 2px solid rgba(0, 0, 0, 0);
          border-radius: 0;
          color: #fff;
          background-color: #00703c;
          box-shadow: 0 2px 0 #002d18;
          text-align: center;
          vertical-align: top;
          cursor: pointer;
          -webkit-appearance: none;
        '
      >Sign in to assess studies</a>

      <h2>Why am I being asked to assess studies?</h2>

      <p>Sponsors or their delegates are asked to review and assess the progress of studies when:</p>

      <ul>
        <li>A study falls behind the agreed milestones</li>
        <li>A study is not recruiting to target</li>
        <li>And the last progress assessment is over three months old</li>
      </ul>

      <p>
        Sponsors or their delegates can request
        <a href='http://localhost:3000/request-support'>NIHR CRN support</a>
        with their research study at any time.
      </p>

      <p>
        <strong>
          Please note that providing regular study assessments is a requirement of the
          <a href='https://www.nihr.ac.uk/documents/researchers/i-need-help-to-deliver-my-research/terms-and-conditions-for-nihr-crn-support.pdf'>Terms and Conditions for NIHR CRN support</a>. Failure to do so may
          result in removal from the NIHR CRN Portfolio.
        </strong>
      </p>

      <h2>Nominate someone else to assess studies</h2>

      <p>
        The NIHR CRN can give other people in your organisation access to this service. Contact
        <a href='mailto:supportmystudy@nihr.ac.uk'>supportmystudy@nihr.ac.uk</a>
        to add a new contact.
      </p>

      <p>
        This email is not monitored, do not reply. If you’re not sure why you’ve received this email or need to be
        removed from our list contact
        <a href='mailto:supportmystudy@nihr.ac.uk'>supportmystudy@nihr.ac.uk</a>.
      </p>
    </div>

    <br />

    <div>
      <table style='width: 540px; border-collapse: collapse'>
        <tbody>
          <tr>
            <td style='width: 162px'>
              <p>
                &nbsp;<img
                  src='https://www.nihr.ac.uk/layout/4.0/assets/external/nihr-logo.png'
                  alt='NIHRLogo'
                  width='206'
                  class='CToWUd'
                  data-bit='iit'
                />
              </p>
            </td>
          </tr>
        </tbody>
      </table>

      <p>
        <a
          href='https://www.nihr.ac.uk/explore-nihr/campaigns/nhs-75/?utm_source=gmail&amp;utm_medium=email&amp;utm_campaign=nhs75&amp;utm_term=signature&amp;utm_content=signature'
          target='_blank'
        ><img
            src='https://www.nihr.ac.uk/layout/img/external/nhs75-shape-the-future-signature.png'
            alt='NIHR, Supporting NHS75. Life changing research funded by the public. Click, get involved in our NHS75 campaign'
            width='540'
            height='101'
            class='CToWUd'
            data-bit='iit'
          />
        </a>
      </p>

      <p>
        Follow us:
        <img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/f_logo_RGB-Blue_58.png'
          alt='FB Logo'
          width='11'
          height='11'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://www.facebook.com/OfficialNIHR' target='_blank'>OfficialNIHR</a>&nbsp;|&nbsp;<img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/Twitter_Logo_Blue.png'
          alt='Twitter logo'
          width='11'
          height='11'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://twitter.com/NIHRresearch' target='_blank'>@NIHRresearch</a>&nbsp;|&nbsp;<img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/LI-In-Bug.png'
          alt='LI Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        /><a href='https://www.linkedin.com/company/nihr-research' target='_blank'>NIHR</a>
        |
        <img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/youtube_logo_rgb_light.png'
          alt='Youtube Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://www.youtube.com/user/NIHRtv' target='_blank'>NIHRtv</a>
        |
        <img
          src='https://www.nihr.ac.uk/layout/img/external/nihr-link-icon.png'
          alt='URL Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        />
        <a
          href='https://www.nihr.ac.uk/?utm_source=gmail&amp;utm_medium=email&amp;utm_campaign=ccsignature'
          target='_blank'
        >nihr.ac.uk</a>
      </p>
      <br />
      <br />
      <p style='color: rgb(34, 34, 34); font-family: arial, sans-serif; background-color: rgb(255, 255, 255)'>
        Confidential information may be contained in this message. If you are not the intended recipient, any reading,
        printing, storage, disclosure, copying or any other action taken in respect of this e-mail is prohibited and may
        be unlawful. If you are not the intended recipient, please notify the sender immediately by using the reply
        function and then permanently delete what you have received.
      </p>
    </div>
  </body>

</html>"
`;

exports[`Email template "contact-assigned.text.hbs" displays correctly 1`] = `
"# NIHR CRN has invited you to assess the progress of your studies

Hello,

You have been nominated as a contact for Mocked org name studies on the NIHR Clinical Research Network's (CRN) research portfolio.

The NIHR CRN supports Sponsors and their delegates to deliver high-quality research, for the benefit of patients, the public and health and care organisations across England.

The NIHR CRN tracks the progress of research studies in its portfolio using data provided by Sponsors or their delegates on behalf of the Department of Health and Social Care.

NIHR CRN asks Sponsors or their delegates to review these studies and provide regular assessments of how the study is progressing.

Please sign in to the Sponsor Engagement Tool to provide updates on studies. (http://localhost:3000/auth/signin)

# Why am I being asked to assess studies?

Sponsors or their delegates are asked to review and assess the progress of studies when:

- A study falls behind the agreed milestones
- A study is not recruiting to target
- And the last progress assessment is over three months old

Sponsors or their delegates can request NIHR CRN support with their research study at any time. (http://localhost:3000/request-support)

Please note that providing regular study assessments is a requirement of the Terms and Conditions for NIHR CRN support.
Failure to do so may result in removal from the NIHR CRN Portfolio. (https://www.nihr.ac.uk/documents/researchers/i-need-help-to-deliver-my-research/terms-and-conditions-for-nihr-crn-support.pdf)

# Nominate someone else to assess studies

The NIHR CRN can give other people in your organisation access to this service. Contact supportmystudy@nihr.ac.uk to add
a new contact.

This email is not monitored, do not reply. If you’re not sure why you’ve received this email or need to be removed from
our list, please contact supportmystudy@nihr.ac.uk.
"
`;

exports[`Email template "contact-removed.html.hbs" displays correctly 1`] = `
"<html lang='en'>
  <body>
    <div>
      <h1>NIHR CRN has removed you as a reviewer for Mocked org name</h1>

      <p>Hello,</p>

      <p>You have been removed as a reviewer of research studies for
        Mocked org name. You will no longer be able to review and assess study progress.
      </p>

      <hr />

      <h2>Not sure why you've been removed?</h2>

      <p>If you think there’s been an error please contact
        <a href='mailto:supportmystudy@nihr.ac.uk'>supportmystudy@nihr.ac.uk</a>.</p>

      <p>If you’re not sure why you’ve received this email or need to be removed from our list contact
        <a href='mailto:supportmystudy@nihr.ac.uk'>supportmystudy@nihr.ac.uk</a>.</p>
    </div>

    <br />

    <div>
      <table style='width: 540px; border-collapse: collapse'>
        <tbody>
          <tr>
            <td style='width: 162px'>
              <p>
                &nbsp;<img
                  src='https://www.nihr.ac.uk/layout/4.0/assets/external/nihr-logo.png'
                  alt='NIHRLogo'
                  width='206'
                  class='CToWUd'
                  data-bit='iit'
                />
              </p>
            </td>
          </tr>
        </tbody>
      </table>

      <p>
        <a
          href='https://www.nihr.ac.uk/explore-nihr/campaigns/nhs-75/?utm_source=gmail&amp;utm_medium=email&amp;utm_campaign=nhs75&amp;utm_term=signature&amp;utm_content=signature'
          target='_blank'
        ><img
            src='https://www.nihr.ac.uk/layout/img/external/nhs75-shape-the-future-signature.png'
            alt='NIHR, Supporting NHS75. Life changing research funded by the public. Click, get involved in our NHS75 campaign'
            width='540'
            height='101'
            class='CToWUd'
            data-bit='iit'
          />
        </a>
      </p>

      <p>
        Follow us:
        <img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/f_logo_RGB-Blue_58.png'
          alt='FB Logo'
          width='11'
          height='11'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://www.facebook.com/OfficialNIHR' target='_blank'>OfficialNIHR</a>&nbsp;|&nbsp;<img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/Twitter_Logo_Blue.png'
          alt='Twitter logo'
          width='11'
          height='11'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://twitter.com/NIHRresearch' target='_blank'>@NIHRresearch</a>&nbsp;|&nbsp;<img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/LI-In-Bug.png'
          alt='LI Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        /><a href='https://www.linkedin.com/company/nihr-research' target='_blank'>NIHR</a>
        |
        <img
          src='https://www.nihr.ac.uk/images/about-us/stay-up-to-date/youtube_logo_rgb_light.png'
          alt='Youtube Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        />
        <a href='https://www.youtube.com/user/NIHRtv' target='_blank'>NIHRtv</a>
        |
        <img
          src='https://www.nihr.ac.uk/layout/img/external/nihr-link-icon.png'
          alt='URL Logo'
          width='11'
          height='9'
          class='CToWUd'
          data-bit='iit'
        />
        <a
          href='https://www.nihr.ac.uk/?utm_source=gmail&amp;utm_medium=email&amp;utm_campaign=ccsignature'
          target='_blank'
        >nihr.ac.uk</a>
      </p>
      <br />
      <br />
      <p style='color: rgb(34, 34, 34); font-family: arial, sans-serif; background-color: rgb(255, 255, 255)'>
        Confidential information may be contained in this message. If you are not the intended recipient, any reading,
        printing, storage, disclosure, copying or any other action taken in respect of this e-mail is prohibited and may
        be unlawful. If you are not the intended recipient, please notify the sender immediately by using the reply
        function and then permanently delete what you have received.
      </p>
    </div>
  </body>

</html>"
`;

exports[`Email template "contact-removed.text.hbs" displays correctly 1`] = `
"# NIHR CRN has removed you as a reviewer for Mocked org name

Hello, You have been removed as a reviewer of research studies for Mocked org name. You will no longer be able to
review and assess study progress.

# Not sure why you've been removed?

If you think there’s been an error please contact supportmystudy@nihr.ac.uk.

If you’re not sure why you’ve received this email or need to be removed from our list contact supportmystudy@nihr.ac.uk."
`;

exports[`Email template "index.spec.ts" displays correctly 1`] = `
"import { EXTERNAL_CRN_TERMS_CONDITIONS_URL, EXTERNAL_CRN_URL, SUPPORT_PAGE } from '../../constants/routes'
import emailTemplates from '.'

const templates = Object.keys(emailTemplates).filter((name) => name !== 'index.js')

test.each(templates)('Email template %p displays correctly', (key) => {
  const template = emailTemplates[key] as (data: Record<string, string>) => string
  expect(
    template({
      crnLink: EXTERNAL_CRN_URL,
      organisationName: 'Mocked org name',
      requestSupportLink: \`http://localhost:3000\${SUPPORT_PAGE}\`,
      signInLink: 'http://localhost:3000/auth/signin',
      termsAndConditionsLink: EXTERNAL_CRN_TERMS_CONDITIONS_URL,
    })
  ).toMatchSnapshot()
})
"
`;
